service-python/assessclaimdc7101/src/lib/bp_calculator.py

```diff
@@ -21,6 +21,30 @@ def sort_bp(bp_readings):
     return bp_readings
 
 
+def deduplicate(bp_readings):
+    """
+    Return bp readings with any lighthouse duplicates removed. A duplicate is identified by having the same diastolic
+    value, systolic value and date. HDR data has organization set to VAMC Other Output Reports.
+    :param bp_readings: full list of BP readings
+    :return: deduplicated list
+    """
+    deduplicated_readings = []
+    for reading in bp_readings:
+        duplicate = False
+        if reading["dataSource"] == "LH":
```

Comment:
We talked about the possibility of using a `set` to de-duplicate this. I might still like to see this implemented to mitigate away the loop-within-a-loop later, but that can be a performance tweak for later.

Reply:
i agree, this needs to use set or dictionary. Although the data sizes are small enough that performance gain will not be noticeable.  I would do it anyway, just better coding.

Reply:
It is not a set though. In the requirements, HDR data should not be removed even if there are duplicates within it.

Reply:
From requirements:

Potential edge case: If there's 2 BP readings in HDR and 1 BP reading in Lighthouse that have the same readings and observation date, do we consider it 1 reading or 2? Going by my assumption that we should not de-duplicate within HDR or within Lighthouse data, that would leave us with 2 readings.

Reply:
Based on the requirements I say this could work. Loop through one and anything not `LH` or `VAMC Other Output Reports` is unique so all thats left is just comparing the (diastolic + systolic + date) in a set for the remaining uniqueness.
```
deduplicated_readings = []
unique_readings = set()
for reading in bp_readings:
    if reading["dataSource"] != "LH":
        deduplicated_readings.append(reading)
        continue
    if reading["organization"] != "VAMC Other Output Reports":
        deduplicated_readings.append(reading)
        continue
    reading_key = (reading["diastolic"]["value"], reading["systolic"]["value"], reading["date"])
    if reading_key not in unique_readings:
        unique_readings.add(reading_key)
        deduplicated_readings.append(reading)
return deduplicated_readings
```

Reply:
This is clever but I do not think it will work because all MAS data will be caught by the first if statement and so it will not be added to `unique readings`. I do not think it would compare MAS data to LH data. It's possible this implementation could work with some more complicated conditions, I am not sure.

Reply:
Is HDR the same as MAS or a 3rd data source?

Reply:
We are getting HDR data from the MAS API. It is identified by having "VAMC Other Output Reports" as the organization.